plugins/importexport/native/filter/NativeXmlPKPPublicationFilter.inc.php

```diff
@@ -0,0 +1,292 @@
+	 * @param $element DOMElement
+	 * @param $publication PKPPublication
+	 */
+	function parseIdentifier($element, $publication) {
+		$deployment = $this->getDeployment();
+		$submission = $deployment->getSubmission();
+
+		$advice = $element->getAttribute('advice');
+		switch ($element->getAttribute('type')) {
+			case 'internal':
+				// "update" advice not supported yet.
+				assert(!$advice || $advice == 'ignore');
+
+				if ($element->textContent == $submission->getData('currentPublicationId')) {
+					$submission->setData('currentPublicationId', $publication->getId());
+				}
+
+				break;
+			case 'public':
+				if ($advice == 'update') {
+					$publication->setData('pub-id::publisher-id', $element->textContent);
+				}
+				break;
+			default:
+				if ($advice == 'update') {
+					$pubIdPlugins = PluginRegistry::loadCategory('pubIds', true, $deployment->getContext()->getId());
+					$publication->setData('pub-id::'.$element->getAttribute('type'), $element->textContent);
+				}
+		}
+	}
+
+	/**
+	 * Parse an authors element
+	 * @param $node DOMElement
+	 * @param $publication PKPPublication
+	 */
+	function parseAuthors($node, $publication) {
+		for ($n = $node->firstChild; $n !== null; $n=$n->nextSibling) {
+			if (is_a($n, 'DOMElement')) {
+				assert($n->tagName == 'author');
+				$this->parseAuthor($n, $publication);
+			}
+		}
+	}
+
+	/**
+	 * Parse an author and add it to the submission.
+	 * @param $n DOMElement
+	 * @param $publication Publication
+	 */
+	function parseAuthor($n, $publication) {
+		$filterDao = DAORegistry::getDAO('FilterDAO'); /** @var $filterDao FilterDAO */
+		$importFilters = $filterDao->getObjectsByGroup('native-xml=>author');
+		assert(count($importFilters)==1); // Assert only a single unserialization filter
+		$importFilter = array_shift($importFilters);
+		$importFilter->setDeployment($this->getDeployment());
+		$authorDoc = new DOMDocument();
+		$authorDoc->appendChild($authorDoc->importNode($n, true));
+		return $importFilter->execute($authorDoc);
+	}
+
+	//
+	// Helper functions
+	//
+	/**
+	 * Get node name to setter function mapping for localized data.
+	 * @return array
+	 */
+	function _getLocalizedPublicationFields() {
+		return array(
+			'title',
+			'prefix',
+			'subtitle',
+			'abstract',
+			'coverage',
+			'type',
+			'source',
+			'rights',
+			'copyrightHolder',
+		);
+	}
+
+	/**
+	 * Get node name to DAO and insert function mapping.
+	 * @return array
+	 */
+	function _getControlledVocabulariesMappings() {
+		return array(
+			'keywords' => array('SubmissionKeywordDAO', 'insertKeywords'),
+			'agencies' => array('SubmissionAgencyDAO', 'insertAgencies'),
+			'disciplines' => array('SubmissionDisciplineDAO', 'insertDisciplines'),
+			'subjects' => array('SubmissionSubjectDAO', 'insertSubjects'),
+		);
+	}
+
+	/**
+	 * Get the representation export filter group name
+	 * @return string
+	 */
+	function getRepresentationExportFilterGroupName() {
+		assert(false); // Subclasses must override
```

Comment:
This could just be an abstract class (we were using assertions here for the sake of older versions of PHP).